azkaban-exec-server/src/main/java/azkaban/execapp/FlowRampManager.java

```diff
@@ -626,23 +629,19 @@ synchronized private boolean isDatabasePushingActionRequired() {
     return ((!isRampPollingServiceEnabled) && (statusPushIntervalMax <= rampDataModel.getEndFlowCount()));
   }
 
-  // This check function is only applied on non-polling modeƒ
+  // This check function is only applied on non-polling mode
   synchronized private boolean isDatabasePullingActionRequired() {
     return ((!isRampPollingServiceEnabled) && (statusPullIntervalMax <= rampDataModel.getBeginFlowCount()));
   }
 
   synchronized private Action convertToAction(Status status) {
-    switch (status) {
-      case FAILED:
-        return Action.FAILED;
-      case SUCCEEDED:
-        return Action.SUCCEEDED;
-      default:
-        return Action.IGNORED;
-    }
+    if (Status.FAILED.equals(status)) return Action.FAILED;
```

Comment:
The switch case is more scalable for future. Any reason why moved to if...else construct?

Reply:
yes. I want to leverage the Status.isStatusSucceeded function. Status is actually a non-ramp object. If the Status.isStatusSucceeded's enum list has been updated later, we definitely would like the ramp feature will carry the change automatically without any other extra change.